Sources/Common/Core/Math/index.js

```diff
@@ -1202,22 +1209,22 @@ export function diagonalize3x3(a_3x3, w_3, v_3x3) {
 
 export function singularValueDecomposition3x3(a_3x3, u_3x3, w_3, vT_3x3) {
   let i;
-  const B = [createArray(3), createArray(3), createArray(3)];
+  const B = createArray(9);
 
   // copy so that A can be used for U or VT without risk
   for (i = 0; i < 3; i++) {
-    B[0][i] = a_3x3[0][i];
-    B[1][i] = a_3x3[1][i];
-    B[2][i] = a_3x3[2][i];
+    B[i] = a_3x3[i];
+    B[3 + i] = a_3x3[3 + i];
+    B[6 + i] = a_3x3[6 + i];
   }
```

Comment:
actually, isn't the following good enough: ? `const B = [...a_3x3]`

you might want to do a pass on all your modified code to make sure you have not let such unneeded complexity around.